tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,776 @@
+# were found in the database                           #
+#======================================================#
+if(keys %$uploadsRef != @uploadID) {
+    foreach(@uploadID) {
+        if(!defined $uploadsRef->{$_}) {
+            printf STDERR "No upload found in table mri_upload with upload ID $_\n";
+            exit $NeuroDB::ExitCodes::INVALID_ARG;
+        }
+    }
+}
+my %tarchiveID = map { $uploadsRef->{$_}->{'TarchiveID'} => 1 } keys %$uploadsRef;
+
+if(keys %tarchiveID != 1) {
+    print STDERR "The upload IDs passed on the command line have different TarchiveIDs: ";
+    print STDERR join(',', keys %tarchiveID);
+    print STDERR ". Aborting\n";
+    exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+# Since they are all the same, we can check the TarchiveID of the first
+# upload
+my $tarchiveID = (keys %tarchiveID)[0];
+if(!$tarchiveID) {
+    print STDERR "No tarchive ID found in table mri_upload for the uploads IDs passed on the command line\n";
+    exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+my $archiveLocation = (values %$uploadsRef)[0]->{'ArchiveLocation'};
+if(!defined $tarchiveLibraryDir && $archiveLocation !~ /^\//) {
+    print STDERR "Cannot determine absolute path for archive '$archiveLocation' "
+        . "since config setting 'tarchiveLibraryDir' is not set. Aborting.\n";
+    exit $NeuroDB::ExitCodes::MISSING_CONFIG_SETTING;
+}
+
+#===================================================#
+# Make sure there are no entries in files_qc_status #
+# and feedback_mri_comments for the files in the    #
+# archive                                           #
+#===================================================#
+
+if(&hasQcOrComment($dbh, $tarchiveID)) {
+    print STDERR "Cannot delete upload(s) passed on command line: there is QC information"
+        . " defined on the MINC files for the associated tarchive (ID=$tarchiveID)\n";
+    exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+
+#=================================================================#
+# Find the absolute paths of all files associated to the          #
+# upload(s) passed on the command lines in tables files           #
+# files_intermediary, parameter_file, mri_protocol_violated_scans #
+# mri_violations_log, MRICandidateErrors and tarchive             #
+#=================================================================#
+my %files;
+$files{'files'}                       = &getFilesRef($dbh, $tarchiveID, $dataDirBasepath);
+$files{'files_intermediary'}          = &getIntermediaryFilesRef($dbh, $tarchiveID, $dataDirBasepath);
+$files{'parameter_file'}              = &getParameterFilesRef($dbh, $tarchiveID, $dataDirBasepath);
+$files{'mri_protocol_violated_scans'} = &getMriProtocolViolatedScansFilesRef($dbh, $tarchiveID, $dataDirBasepath);
+$files{'mri_violations_log'}          = &getMriViolationsLogFilesRef($dbh, $tarchiveID, $dataDirBasepath);
+$files{'MRICandidateErrors'}          = &getMRICandidateErrorsFilesRef($dbh, $tarchiveID, $dataDirBasepath);
```

Comment:
get the processing protocol files (ProcessProtocolID) = mri_processing_protocol (ProcessProtocolID) and make sure that no other entries in the files table are using the same processing protocol.

Reply:
Done.